src/cluster/cluster.go

```diff
@@ -60,18 +60,33 @@ func (c *Cluster) UpdatePeers(urlPath string, message PeerUpdateMessage, exclude
 	return nil
 }
 
+//NewDevice -
+func (c *Cluster) NewDevice(host string, port string) (device.Device, error) {
+	dvc := device.Device{
+		ID:   c.generateUniqueID(),
+		Host: host,
+		Port: port,
+	}
+	err := dvc.LoadSolenoids()
```

Comment:
This area is probably going to end up having more functions like 'LoadSerial' or 'LoadXIOComponent'. do you think we should pull this into a 'LoadComponents' type function now? or wait till we need it?

Reply:
I just stubbed in igniters and added a components interface so they can be listed together. I think we can wait a moment and I'll update this name int the next PR